File: UI/tendergo_desktop/lib/shared/models/requests/category_insert_request.dart
class CategoryInsertRequest {
  final String name;
  final String description;

  CategoryInsertRequest({
    required this.name,
    required this.description,
  });

  String? validate() {
    if (name.trim().isEmpty) {
      return 'Naziv kategorije je obavezan.';
    }

    if (description.trim().isEmpty) {
      return 'Opis kategorije je obavezan.';
    }

    return null;
  }

  Map<String, dynamic> toJson() => {
        'name': name,
        'description': description,
      };
}

File: backend/TenderGo.Models/Requests/CategoryInsertRequest.cs
using System.ComponentModel.DataAnnotations;

public class CategoryInsertRequest
{
    [Required(ErrorMessage = "Naziv kategorije je obavezan.")]
    [StringLength(100)]
    public string Name { get; set; } = string.Empty;

    [Required(ErrorMessage = "Opis kategorije je obavezan.")]
    [StringLength(500)]
    public string Description { get; set; } = string.Empty;
}